frb_codegen/src/generator/dart/ty_optional.rs

```diff
@@ -1,20 +1,29 @@
 use crate::generator::dart::ty::*;
 use crate::ir::*;
+use crate::target::Acc;
 use crate::type_dart_generator_struct;
-use crate::utils::BlockIndex;
 
 type_dart_generator_struct!(TypeOptionalGenerator, IrTypeOptional);
 
 impl TypeDartGeneratorTrait for TypeOptionalGenerator<'_> {
-    fn api2wire_body(&self, _block_index: BlockIndex) -> Option<String> {
-        Some(format!(
-            "return raw == null ? ffi.nullptr : _api2wire_{}(raw);",
-            self.ir.inner.safe_ident()
-        ))
+    fn api2wire_body(&self) -> Acc<Option<String>> {
+        Acc {
+            io: Some(format!(
+                "return raw == null ? ffi.nullptr : api2wire_{}(raw);",
+                self.ir.inner.safe_ident()
+            )),
+            wasm: self.context.wasm().then(|| {
+                format!(
+                    "return raw == null ? null : api2wire_{}(raw);",
+                    self.ir.inner.safe_ident()
+                )
+            }),
```

Comment:
uber-nit: The two branches only diff in `nullptr` vs `null`, so maybe extract a anonymous closure to do this, instead of dup code